File: problem-solving/_lc-27-easy-remove-element/1-solution.py
"""
https://leetcode.com/problems/remove-element/

Complexity
----------
Time: O(n)
Space: O(1)
"""

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i = 0
        for x in nums:
            if x != val:
                nums[i] = x
                i += 1
        return i

File: problem-solving/lc-75-medium-sort-colors/1-solution.go
/*
https://leetcode.com/problems/sort-colors/

Complexity
----------
Time: O(n)
Space: O(1)
*/

func sortColors(nums []int)  {
    low, mid := 0, 0
    high := len(nums) - 1
    for mid <= high {
        if nums[mid] == 0 {
            nums[low], nums[mid] = nums[mid], nums[low]
            low += 1
            mid += 1
        } else if nums[mid] == 2 {
            nums[mid], nums[high] = nums[high], nums[mid]
            high -= 1
        } else {
            mid += 1
        }
    }
}
